clang/test/ModuleBuildDaemon/handshake.c

```diff
@@ -0,0 +1,25 @@
+// Check that a clang invocation can spawn and handshake with a module build daemon
+
+// RUN: %kill-process "-cc1modbuildd mbd-handshake"
+// RUN: rm -rf mbd-handshake %t
+// RUN: split-file %s %t
+
+//--- main.c
+int main() {return 0;}
+
+// RUN: %clang -fmodule-build-daemon=mbd-handshake -Rmodule-build-daemon %t/main.c &> %t/output-new || true
```

Comment:
The daemon exits after 15 seconds of waiting for a new connection now.

Reply:
I don't think the current timeout implementation will work for a more complicated case than just a handshake (happy to prove wrong with an example).

For situation like: client sends handshake, server ack before timeout, client should send payload but died or killed, server is forever waiting for the payload.

Reply:
Right, it would be great to land the first version with a little more than just a hand shake. Maybe show a case with the situation I mentioned above (with a simple payload and response transferred after the handshake). Then I will feel more confident about the design.

Reply:
That's understandable. To ensure I understand, do you want to see an example where the client dies after receiving the handshake response and fails to deliver a payload or do you want to see a payload and response sent back and forth after the handshake? I think it would be sufficient to carry out a second handshake if it's the latter. The only difference between a second handshake and a new payload message is the contents of the buffer, which I don't think is super important since we mainly care about the mechanisms behind preventing zombie processes.

Reply:
I don't need to see an example or a testcase as I understand that would might be hard to write. I just want to see the implementation of the daemon that does something after the handshake (another handshake is fine) so I can reason if the daemon is leaked if the client crashed in the middle.